ambry-messageformat/src/main/java/com/github/ambry/messageformat/MessageFormatSend.java

```diff
@@ -118,57 +119,49 @@ private void fetchDataFromReadSet() throws MessageFormatException {
       logger.trace("Calculate offsets of messages for one partition, MessageFormatFlag : {} number of messages : {}",
           flag, messageCount);
       for (int i = 0; i < messageCount; i++) {
-        if (flag == MessageFormatFlags.All) {
+        if (flag == MessageFormatFlags.All || flag == MessageFormatFlags.Blob) {
           // just copy over the total size and use relative offset to be 0
           // We do not have to check any version in this case as we dont
           // have to read any data to deserialize anything.
-          sendInfoList.add(i, new SendInfo(0, readSet.sizeInBytes(i)));
-          messageMetadataList.add(i, null);
-          totalSizeToWrite += readSet.sizeInBytes(i);
           readSet.doPrefetch(i, 0, readSet.sizeInBytes(i));
+
+          if (flag == MessageFormatFlags.All) {
+            sendInfoList.add(i, new SendInfo(0, readSet.sizeInBytes(i)));
+            messageMetadataList.add(i, null);
+            totalSizeToWrite += readSet.sizeInBytes(i);
+          } else if (flag == MessageFormatFlags.Blob) {
+            ByteBuf blobAll = readSet.getPrefetchedData(i);
+            InputStream is = new ByteBufInputStream(blobAll);
```

Comment:
nit: we can use a more descriptive name than just `is`, something like `'blobInputStream`.

Reply:
sure. Changed.